### PR TITLE
fix for grant start date scopes

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

The start date after and start date before grant scopes were excluding too many grants. The scope has been modified to return all the recipients with grants active during the time frame. An end date scope was added to duplicate the start date functionality so it would show grants during the same time frame (I think that makes sense).

Also fixed an error where date "is" was missing a default value.

## How to test

View the overview widget. Make sure **start date before** & **after** and **end date before** & **after** return sensible numbers. Also double check the scopes and make sure those make logical sense.

On a fresh page, add a date started -> is -> last thirty days filter. Should work, not say "Invalid date"

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-697
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-699

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/scopes/grants/index.js [hunk 1]
```diff
@@ -29,8 +29,14 @@ export const topicToQuery = {
     ctn: (query) => withStateCode(query),
   },
   startDate: {
-    bef: (query) => beforeStartDate(query),
-    aft: (query) => afterStartDate(query),
+    bef: (query) => activeBefore(query),
```
Comment (jasalisbury) [inline src/scopes/grants/index.js:32]: Maybe this is a sign we need to refactor filters/scopes a bit, but I _could_ see a future where we want to filter grants by their start date. The way `activeBefore` works means if we tried to reuse this filter to actually filter grants by start date we will get the wrong result. I'd hate for a developer in the future to ask for `grants` with a start date before x date and get back `grants` with an end date before x.

Also we could probably use a comment on why we have a `startDate` filter that actually filters on `endDate`. Took me a while to understand what was going on, maybe my brain has yet to fully turned on today!

File: src/scopes/grants/startDate.js [hunk 2]
```diff
@@ -1,19 +1,32 @@
...
+
+  return {
+    [Op.or]: scopes,
   };
 }
 
-export function afterStartDate(date) {
-  return {
-    [Op.and]: {
-      [Op.or]: compareDate(date, 'startDate', Op.gt),
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply src/scopes/grants/startDate.js:15]: no, the other places where compareDate is used should continue to work as they have been

File: src/scopes/grants/index.js (hunk 1 above)
Comment (thewatermethod) [reply src/scopes/grants/index.js]: I just pushed up a proposed solution, there's some comments elaborating, but basically tried to add an options object to filterToScopes that allows for different filter options to be returned and can be built upon as we go

**Comment by jasalisbury** [comment]
Do you think this PR could also address https://ocio-jira.acf.hhs.gov/browse/TTAHUB-699? Not sure of the lift, might be a quick fix. If not we can just fix it in a separate PR.

File: frontend/yarn-audit-known-issues [hunk 3]
```diff
@@ -1,5 +1,5 @@
...at checked for `..` within the normalized and split portions of the path.\n\nThis only affects users of `node-tar` on Windows systems.\n\n### Patches\n\n4.4.18 || 5.0.10 || 6.1.9\n\n### Workarounds\n\nThere is no reasonable way to work around this issue without performing the same path normalization procedures that node-tar now does.\n\nUsers are encouraged to upgrade to the latest patched versions of node-tar, rather than attempt to sanitize paths themselves.\n\n### Fix\n\nThe fixed versions strip path roots from all paths prior to being resolved against the extraction target folder, even if such paths are not \"absolute\".\n\nAdditionally, a path starting with a drive letter and then two dots, like `c:../`, would bypass the check for `..` path portions.  This is checked properly in the patched versions.\n\nFinally, a defense in depth check is added, such that if the `entry.absolute` is outside of the extraction taret, and we are not in preservePaths:true mode, a warning is raised on that entry, and it is skipped.  Currently, it is believed that this check is redundant, but it did catch some oversights in development.\n","url":"https://github.com/advisories/GHSA-5955-9wpr-37jh"}}}
```
Comment (jasalisbury) [inline frontend/yarn-audit-known-issues:1]: Pretty annoying we have to keep updating this. `react-js-pagination` is when depends on tar. Looking at the repo it never actually uses tar, just depends on it. Also not sure why pagination on the frontend would ever need to use tar...

Comment (thewatermethod) [reply frontend/yarn-audit-known-issues:1]: npm is a trap

Comment (jasalisbury) [inline frontend/yarn-audit-known-issues:1]: I'd make sure this file goes away after you merge main into this branch

**Review by jasalisbury — APPROVED** [review]
:shipit: 
> Fix for this is included here, btw

Thanks!

**Pull request merged**